### PR TITLE
[AEM-139] Use wcm-io Maven dependencies

### PR DESCRIPTION
**Pull request by matiasvigil**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -20,57 +20,38 @@
...
-        <sling.version>1.2.2</sling.version>
-        <geronimo.version>1.0</geronimo.version>
+        <io.wcm.aem.version>6.4.2.0000</io.wcm.aem.version>
+        <io.wcm.aem.mock.version>2.5.0</io.wcm.aem.mock.version>
         <acs-commons.version>3.19.0</acs-commons.version>
-        <servlet.api.version>3.1.0</servlet.api.version>
-        <jsp.api.version>2.1</jsp.api.version>
-        <jcr.version>2.0</jcr.version>
         <taglib.version>5.7.4</taglib.version>
-        <guava.version>22.0</guava.version>
```
Comment (wrandall22) [inline pom.xml]: We'll still need this. Guava 15 won't work for some of our projects, which is what wcm-io is bringing in.

File: pom.xml [hunk 2]
```diff
@@ -110,22 +69,6 @@
                 <scope>provided</scope>
             </dependency>
 
-            <!-- Guava -->
```
Comment (wrandall22) [inline pom.xml]: We'll still need this. Guava 15 won't work for some of our projects, which is what wcm-io is bringing in.

File: pom.xml [hunk 3]
```diff
@@ -146,7 +89,8 @@
                 <scope>provided</scope>
             </dependency>
 
-            <!-- OSGi Dependencies -->
+            <!-- OSGi DEPREACTED Dependencies -->
```
Comment (wrandall22) [inline pom.xml]: Not a big deal, but `deprecated` is misspelled here.

File: pom.xml (hunk 1 above)
Comment (matiasvigil) [reply pom.xml]: Done, I added the Guava 22

File: pom.xml (hunk 2 above)
Comment (matiasvigil) [reply pom.xml]: Done, I added the Guava 22

File: pom.xml (hunk 3 above)
Comment (matiasvigil) [reply pom.xml]: Done

File: pom.xml [hunk 4]
```diff
@@ -20,57 +20,39 @@
...
+        <io.wcm.aem.version>6.4.2.0000</io.wcm.aem.version>
+        <io.wcm.aem.mock.version>2.5.0</io.wcm.aem.mock.version>
         <acs-commons.version>3.19.0</acs-commons.version>
-        <servlet.api.version>3.1.0</servlet.api.version>
-        <jsp.api.version>2.1</jsp.api.version>
-        <jcr.version>2.0</jcr.version>
         <taglib.version>5.7.4</taglib.version>
         <guava.version>22.0</guava.version>
-        <apache.commons.lang.version>3.5</apache.commons.lang.version>
         <jackson.version>2.8.4</jackson.version>
```
Comment (wrandall22) [inline pom.xml:28]: It looks like `io.wcm.maven.aem-dependencies` includes jackson, and we could use the version from there. (see [here](https://github.com/wcm-io/wcm-io-tooling/blob/master/maven/aem-dependencies/6.4.2/pom.xml#L268))

File: pom.xml [hunk 5]
```diff
@@ -146,7 +98,8 @@
                 <scope>provided</scope>
             </dependency>
 
-            <!-- OSGi Dependencies -->
+            <!-- OSGi deprecated Dependencies -->
```
Comment (wrandall22) [inline pom.xml]: I believe we should use this version as an opportunity to remove Felix dependencies as the separate projects begin using this version of `aem-bom`.

Comment (mattdrees) [reply pom.xml]: good idea

File: pom.xml (hunk 4 above)
Comment (matiasvigil) [reply pom.xml:28]: Yes in github includes jackson but check this (see [here](http://central.maven.org/maven2/io/wcm/maven/io.wcm.maven.aem-dependencies/6.4.2.0000/io.wcm.maven.aem-dependencies-6.4.2.0000.pom)) the release version does not containts the jackson dependency

File: pom.xml (hunk 5 above)
Comment (matiasvigil) [reply pom.xml]: I removed the Felix dependencies from bom. I also refactored cru project removing Felix and using the new OSGi annotations.

**Review by wrandall22 — APPROVED** [review]

**Pull request merged**